bats/tests/helpers/utils.bash

```diff
@@ -442,9 +442,20 @@ foreach_k3s_version() {
 
 _foreach_k3s_version() {
     local RD_KUBERNETES_PREV_VERSION=$1
+    local skip_kubernetes_version
+    skip_kubernetes_version=$(cat "$BATS_FILE_TMPDIR/skip-kubernetes-version" 2>/dev/null || echo none)
+    if [[ $skip_kubernetes_version == "$RD_KUBERNETES_PREV_VERSION" ]]; then
+        skip "All remaining tests for Kubernetes $RD_KUBERNETES_PREV_VERSION are skipped"
+    fi
     "$2"
 }
 
+# Tests can call skip_k3s_version to skip the rest of the tests within
+# foreach_k3s_version.
+skip_k3s_version() {
+    echo "$RD_KUBERNETES_PREV_VERSION" >"$BATS_FILE_TMPDIR/skip-kubernetes-version"
+}
```

Comment:
All our `skip_*` functions follow the pattern of calling `skip` internally. It means that the test will `exit` during this call and not return. Could we do this here too?

```suggestion
skip_k3s_version() {
    echo "$RD_KUBERNETES_PREV_VERSION" >"${BATS_FILE_TMPDIR}/skip-kubernetes-version"
    skip "${1:-Skipping all further tests for Kubernetes $RD_KUBERNETES_PREV_VERSION.}"
}
```

Minor note: I try to use brackets when a variable is interpolated in a string, and not just quoted by itself. So I would use `"$FOO"`, but `"${FOO}/bar"`. Of course just a suggestion since we can't enforce it with `make lint` yet. 😄